### PR TITLE
1712-stepper-bug-that-causes-frontend-crash

### PR DESCRIPTION
**Pull request by hanscau**
### Description

Fix second statement being a ReturnStatement in a BlockStatement. The first statement should be reduced before going to the return statement. 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code quality improvements

### How to test

Tests already exists, the expected case was just wrong. 

### Checklist

- [x] I have tested this code

### CODE REVIEW

**Review by martin-henz — APPROVED** [review]
hard to follow the complex logic of reduction of blocks. This PR is not making it worse, and is fixing the given issue. So: best we merge this. (Eventually we should consider simplifying the stepper design.)

**Comment by FYL2003** [comment]
Just some ideas: maybe breaking the big chunk of code into small functional pieces(each function into separate ts files) and add more comment on how the code words would help people better understand the code base.
